### PR TITLE
[StructuralMechanicsApplication] Minor surface load clean up

### PR DESCRIPTION
**Pull request by loumalouomega**
This PR reduces some code duplication (using external utilities,etc...)

I change this looking to the origin of #4272

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_conditions/surface_load_condition_3d.cpp [hunk 1]
```diff
@@ -121,23 +122,21 @@ void SurfaceLoadCondition3D::CalculateAndSubKp(
...
             coeff = Pressure * rN[i] * rDN_De(j, 0) * Weight;
+            noalias(Kij) = coeff * cross_tangent_eta;
 
-            noalias(Kij) -= coeff * cross_tangent_eta;
+            coeff = Pressure * rN[i] * rDN_De(j, 1) * Weight;
+            noalias(Kij) -= coeff * cross_tangent_xi;
 
-            // NOTE TAKE CARE: the load correction matrix should be SUBTRACTED not added
-            MathUtils<double>::SubtractMatrix(rK, Kij, RowIndex, column_index);
+            MathUtils<double>::AddMatrix(rK, Kij, row_index, column_index);
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_conditions/surface_load_condition_3d.cpp:139]: You sure abt this one?
The comment explicitly says not to do this

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_conditions/surface_load_condition_3d.cpp:139]: I changed the order of the other ones

File: applications/StructuralMechanicsApplication/custom_conditions/surface_load_condition_3d.cpp [hunk 2]
```diff
@@ -121,23 +122,21 @@ void SurfaceLoadCondition3D::CalculateAndSubKp(
...
-        const std::size_t RowIndex = i * 3;
+        const std::size_t row_index = i * 3;
         for (std::size_t j = 0; j < number_of_nodes; ++j) {
             const std::size_t column_index = j * 3;
 
-            coeff = Pressure * rN[i] * rDN_De(j, 1) * Weight;
-            noalias(Kij) = coeff * cross_tangent_xi;
-
             coeff = Pressure * rN[i] * rDN_De(j, 0) * Weight;
+            noalias(Kij) = coeff * cross_tangent_eta;
```
Comment (loumalouomega) [inline applications/StructuralMechanicsApplication/custom_conditions/surface_load_condition_3d.cpp:134]: See?, the order is inverted

**Comment by RiccardoRossi** [comment]
but the sign is changed no?

**Comment by loumalouomega** [comment]
> but the sign is changed no?

At the end is the same, just with a clearer order (in my opinion), and using Add instead of Substract

**Review by philbucher — APPROVED** [review]
approving since you seems to be sure what you are doing :)

**Comment by philbucher** [comment]
This makes two tests fail:
~~~
======================================================================
FAIL: test_execution (structural_mechanics_test_factory.ShellT3ThinOrthotropicLaminateLinearStaticTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/philippb/software/Kratos_Master/applications/StructuralMechanicsApplication/tests/structural_mechanics_test_factory.py", line 72, in test_execution
    self.test.RunSolutionLoop()
  File "/home/philippb/software/Kratos_Master/kratos/python_scripts/analysis_stage.py", line 66, in RunSolutionLoop
    self.FinalizeSolutionStep()
  File "/home/philippb/software/Kratos_Master/kratos/python_scripts/analysis_stage.py", line 141, in FinalizeSolutionStep
    process.ExecuteFinalizeSolutionStep()
  File "/home/philippb/software/Kratos_Master/kratos/python_scripts/from_json_check_result_process.py", line 126, in ExecuteFinalizeSolutionStep
    self.assertTrue(isclosethis, msg=(str(value) + " != " + str(value_json) + ", rel_tol = " + str(reltol) + ", abs_tol = " + str(tol) + " : Error checking node " + str(node.Id) + " " + variable_name + " results."))
AssertionError: False is not true : 0.1655320496867231 != 0.17566033683004809, rel_tol = 1e-06, abs_tol = 0.001 : Error checking node 1 DISPLACEMENT_Z results.

======================================================================
FAIL: test_execution (structural_mechanics_test_factory.ShellT3ThickOrthotropicLaminateLinearStaticTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/philippb/software/Kratos_Master/applications/StructuralMechanicsApplication/tests/structural_mechanics_test_factory.py", line 72, in test_execution
    self.test.RunSolutionLoop()
  File "/home/philippb/software/Kratos_Master/kratos/python_scripts/analysis_stage.py", line 66, in RunSolutionLoop
    self.FinalizeSolutionStep()
  File "/home/philippb/software/Kratos_Master/kratos/python_scripts/analysis_stage.py", line 141, in FinalizeSolutionStep
    process.ExecuteFinalizeSolutionStep()
  File "/home/philippb/software/Kratos_Master/kratos/python_scripts/from_json_check_result_process.py", line 126, in ExecuteFinalizeSolutionStep
    self.assertTrue(isclosethis, msg=(str(value) + " != " + str(value_json) + ", rel_tol = " + str(reltol) + ", abs_tol = " + str(tol) + " : Error checking node " + str(node.Id) + " " + variable_name + " results."))
AssertionError: False is not true : 0.15414751002669388 != 0.16315953759223334, rel_tol = 1e-06, abs_tol = 0.001 : Error checking node 1 DISPLACEMENT_Z results.

----------------------------------------------------------------------
~~~

These tests are in the validation-suite, did you not run it?

In any case, can you please have a look?

**Comment by philbucher** [comment]
Ping

**Comment by loumalouomega** [comment]
Ups, I forgot

**Comment by loumalouomega** [comment]
Solved @philbucher

**Pull request merged**